core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -39,18 +39,32 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Sets;
 import org.junit.Assert;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+@RunWith(Parameterized.class)
 public class TestRowDelta extends V2TableTestBase {
+
+  private final String branch;
+
+  @Parameterized.Parameters(name = "branch = {0}")
+  public static Object[] parameters() {
+    return new Object[][] {
+      new Object[] {"main"}, new Object[] {"testBranch"},
+    };
+  }
+
+  public TestRowDelta(String branch) {
+    this.branch = branch;
+  }
+
   @Test
   public void testAddDeleteFile() {
-    table
-        .newRowDelta()
-        .addRows(FILE_A)
-        .addDeletes(FILE_A_DELETES)
-        .addDeletes(FILE_B_DELETES)
-        .commit();
+    SnapshotUpdate rowDelta =
+        table.newRowDelta().addRows(FILE_A).addDeletes(FILE_A_DELETES).addDeletes(FILE_B_DELETES);
 
-    Snapshot snap = table.currentSnapshot();
+    commit(table, rowDelta, branch);
```

Comment:
Odd spacing here. I'd expect the commit to come after rowDelta insteadd of being combined with the latestSnapshot call.